src/FM.LiveSwitch.Mux/Muxer.cs

```diff
@@ -28,201 +33,223 @@ public Muxer(MuxOptions options, ILoggerFactory loggerFactory)
 
         public async Task<bool> Run()
         {
-            if (Options.InputPath == null)
-            {
-                Options.InputPath = Environment.CurrentDirectory;
-                _Logger.LogInformation("Input path defaulting to: {InputPath}", Options.InputPath);
-            }
-
-            if (Options.OutputPath == null)
+            while (true)
```

Comment:
@rken-FM

1. I would move `delay` and `numTries` to local variables since they aren't used outside of the `Run` method.
2. To make it clear that these two variables are related, and specifically related to a retry on failure, I would rename them to `retryDelay` and `retryCount`.
3. This conflicts with the existing `retryDelay` variable, which I think can be removed. Having a 0ms retry on the first failure followed by 200ms thereafter is more complex than a simple 200ms retry every time and doesn't provide the user with any real benefit, especially since this is a rare edge case.
4. I would rename `maxTries` to `retryCountMax` to bind the semantics of it tightly to `retryCount`.
5. The exception log message should provide more detail about the context of what went wrong. Include information like whether we are going to retry, which attempt we are on, etc.
6. The increment of `retryCount` (`numTries`) should be in the `catch` block, no?

```csharp
var retryDelay = 200; // milliseconds
var retryCount = 0;
var retryCountMax = 10000 / retryDelay; // 10 seconds
while (true)
{
    try
    {
        ...
    }
    catch (FileNotFoundException ex)
    {
        if (retryCount >= retryCountMax)
        {
            _Logger.LogError($"A temporary exception was encountered, but retries have been exhausted.", ex);
            throw;
        }

        retryCount++;
        _Logger.LogWarning($"A temporary exception was encountered. Will retry after {retryDelay} milliseconds (attempt #{retryCount}).", ex);
        await Task.Delay(retryDelay).ConfigureAwait(false);
    }
}
```

Reply:
Okay changes have been made, thanks for the inputs.